src/Middleware/HttpOverrides/src/IPNetwork.cs

```diff
@@ -83,21 +93,159 @@ private byte[] CreateMask()
         return mask;
     }
 
-    private static void CheckPrefixLengthRange(IPAddress prefix, int prefixLength)
+    private static bool IsValidPrefixLengthRange(IPAddress prefix, int prefixLength)
     {
         if (prefixLength < 0)
         {
-            throw new ArgumentOutOfRangeException(nameof(prefixLength));
+            return false;
         }
 
         if (prefix.AddressFamily == AddressFamily.InterNetwork && prefixLength > 32)
         {
-            throw new ArgumentOutOfRangeException(nameof(prefixLength));
+            return false;
         }
 
         if (prefix.AddressFamily == AddressFamily.InterNetworkV6 && prefixLength > 128)
         {
-            throw new ArgumentOutOfRangeException(nameof(prefixLength));
+            return false;
+        }
+
+        return true;
+    }
+
+    /// <summary>
+    /// Converts the specified <see langword="string"/> representation of an IP address
+    /// and a prefix length to its <see cref="IPNetwork"/> equivalent.
+    /// </summary>
+    /// <param name="networkString">The <see langword="string"/> to convert, in CIDR notation.</param>
+    /// <returns>
+    /// The <see cref="IPNetwork"/> equivalent to the IP address and prefix length contained in <paramref name="networkString"/>.
+    /// </returns>
+    /// <exception cref="ArgumentNullException"><paramref name="networkString"/> is <see langword="null"/>.</exception>
+    /// <exception cref="FormatException"><paramref name="networkString"/> is not in the correct format.</exception>
+    /// <exception cref="ArgumentOutOfRangeException">The prefix length contained in <paramref name="networkString"/> is out of range.</exception>
+    /// <inheritdoc cref="TryParseComponents(ReadOnlySpan{char}, out IPAddress?, out int)"/>
+    public static IPNetwork Parse(string? networkString)
```

Comment:
Is this method needed?
string is convertable to ROS<char> implicitely by callers too, so this overload could be stripped.

Or is it here because of the null-check?

Reply:
That's a good point. I'll go ahead and remove these `string?` overloads to simplify things.

Out of curiosity, do we always prefer only defining methods which accept ROS rather than defining overloads for both ROS and string? Originally, I added this overload so I could include the null check on `networkString`, although I'm not sure if throwing an `ArgumentNullException` here would give us any practical benefits vs. just converting a null string to ROS and then ending up with a `FormatException` later on in the method.